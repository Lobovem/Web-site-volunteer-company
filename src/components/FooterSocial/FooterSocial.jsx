import s from './FooterSocial.module.scss';

export const FooterSocial = () => {
  return (
    <div className={s.social}>
      <div className={s.social__wrap}>
        <h2 className={s.social__title}>соціальні мережі</h2>
        <a className={s.social__wrapLink} target="blank" href="https://www.instagram.com/ukrhumbat/?igshid=YmMyMTA2M2Y=">
          <img className={s.social__img} src="./img/icon-instagram.svg" alt="icon-instagram.svg" />
          <p className={s.social__desc}>instagram</p>
        </a>

        <a
          className={s.social__wrapLink}
          target="blank"
          href="https://www.facebook.com/people/%D0%9C%D0%91%D0%A4-%D0%A3%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D1%81%D1%8C%D0%BA%D0%B8%D0%B9-%D0%B3%D1%83%D0%BC%D0%B0%D0%BD%D1%96%D1%82%D0%B0%D1%80%D0%BD%D0%B8%D0%B9-%D0%B1%D0%B0%D1%82%D0%B0%D0%BB%D1%8C%D0%BE%D0%BD/100085936770532/"
        >
          <img className={s.social__img} src="./img/icon-facebook.svg" alt="icon-facebook.svg" />
          <p className={s.social__desc}>facebook</p>
        </a>
      </div>
    </div>
  );
};
