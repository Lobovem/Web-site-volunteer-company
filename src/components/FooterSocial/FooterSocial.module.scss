@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.social {

  flex: 1;

  &__wrap {
    margin-bottom: 50px;
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-24;
    line-height: 24px;
    text-transform: uppercase;
    color: $color-white;
    mix-blend-mode: normal;
    margin: 0;
    padding: 0;
    margin-bottom: 30px;
  }

  &__desc {
    font-weight: $font-weight-500;
    font-size: $font-size-14;
    line-height: 17px;
    mix-blend-mode: normal;
    text-decoration: none;
    display: block;

  }

  &__wrapLink {
    position: relative;
    margin-bottom: 20px;
    padding-left: 40px;
    display: flex;
    color: $color-white;

    align-items: center;
    text-decoration: none;
    transition: all 0.3s ease-in-out;

    &:hover {
      color: $color-hover;
    }
  }

  &__facebookBox {
    position: relative;
    padding-left: 40px;

  }

  &__img {
    position: absolute;
    top: 50%;
    left: 0;
    transform: translate(0%, -50%);
  }
}