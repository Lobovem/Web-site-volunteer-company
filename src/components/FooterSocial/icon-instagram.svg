<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14.0346 0.462891H5.95501C2.6714 0.462891 0 3.03163 0 6.18904V13.9581C0 17.1155 2.6714 19.6842 5.95501 19.6842H14.0346C17.3182 19.6842 19.9896 17.1155 19.9896 13.9581V6.18904C19.9896 3.03163 17.3182 0.462891 14.0346 0.462891ZM17.9786 13.9581C17.9786 16.0526 16.2128 17.7506 14.0346 17.7506H5.95501C3.77678 17.7506 2.01095 16.0526 2.01095 13.9581V6.18904C2.01095 4.09449 3.77678 2.39656 5.95501 2.39656H14.0346C16.2128 2.39656 17.9786 4.09449 17.9786 6.18904V13.9581Z" fill="white"/>
<path d="M9.99521 5.10254C7.14446 5.10254 4.8252 7.33267 4.8252 10.0738C4.8252 12.815 7.14446 15.0451 9.99521 15.0451C12.846 15.0451 15.1652 12.815 15.1652 10.0738C15.1652 7.33263 12.846 5.10254 9.99521 5.10254ZM9.99521 13.1115C8.25049 13.1115 6.83615 11.7515 6.83615 10.0739C6.83615 8.3962 8.25053 7.03621 9.99521 7.03621C11.7399 7.03621 13.1543 8.3962 13.1543 10.0739C13.1543 11.7515 11.7399 13.1115 9.99521 13.1115Z" fill="white"/>
<path d="M15.1744 6.33071C15.8586 6.33071 16.4132 5.79738 16.4132 5.13948C16.4132 4.48158 15.8586 3.94824 15.1744 3.94824C14.4902 3.94824 13.9355 4.48158 13.9355 5.13948C13.9355 5.79738 14.4902 6.33071 15.1744 6.33071Z" fill="white"/>
</svg>
