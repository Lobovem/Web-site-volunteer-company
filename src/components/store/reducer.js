import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
