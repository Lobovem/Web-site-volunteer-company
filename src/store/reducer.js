import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  todoListState: [],
};

export const appReducer = createSlice({
  name: 'todoList',
  initialState: initialState,
  reducers: {
    // add_todo: (state, action) => {
    //   state.todoListState = [
    //     ...state.todoListState,
    //     {
    //       id: Math.random(),
    //       task: action.payload,
    //       complete: false,
    //       isEditing: false,
    //     },
    //   ];
    // },
    // delete_todo: (state, action) => {
    //   state.todoListState = state.todoListState.filter((todo) => todo.id !== action.payload.id);
    // },
  },
});

// export const userInputSelector = (state) => state.todoList.inputState;
// export const todoListSelector = (state) => state.todoList.todoListState;

// export const { add_todo } = todoReducer.actions;
export default appReducer.reducer;
