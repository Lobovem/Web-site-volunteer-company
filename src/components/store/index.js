// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
