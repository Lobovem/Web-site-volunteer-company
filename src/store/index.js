// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
