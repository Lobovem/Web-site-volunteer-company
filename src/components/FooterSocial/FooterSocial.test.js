import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { FooterSocial } from './FooterSocial';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';

describe('FooterSocial` component', () => {
  test('renders FooterSocial component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <Router>
          <FooterSocial />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('should render the title (H2) component social', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterSocial />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { level: 2 });
    expect(titleSection).toBeInTheDocument();
  });

  test('should render the title component social', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterSocial />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { name: /СОЦІАЛЬНІ МЕРЕЖІ/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('renders FooterSocial component with 2 images (link elements)', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterSocial />
        </Router>
      </Provider>
    );
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(2);
  });

  test('renders and search elements links', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterSocial />
        </Router>
      </Provider>
    );
    const links = screen.getAllByRole('link'); // Search all links
    expect(links).toHaveLength(2);

    const linkSocial1 = screen.getByText('instagram'); // Search link about text
    expect(linkSocial1).toBeInTheDocument(); // Check out link in DOM

    const linkSocial2 = screen.getByText('facebook');
    expect(linkSocial2).toBeInTheDocument();
  });
});
