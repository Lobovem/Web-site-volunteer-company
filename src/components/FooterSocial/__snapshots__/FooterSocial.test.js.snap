// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterSocial\` component renders FooterSocial component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="social"
      >
        <div
          class="social__wrap"
        >
          <h2
            class="social__title"
          >
            соціальні мережі
          </h2>
          <a
            class="social__wrapLink"
            href="https://www.instagram.com/ukrhumbat/?igshid=YmMyMTA2M2Y="
            target="blank"
          >
            <img
              alt="icon-instagram.svg"
              class="social__img"
              src="/img/icon-instagram.svg"
            />
            <p
              class="social__desc"
            >
              instagram
            </p>
          </a>
          <a
            class="social__wrapLink"
            href="https://www.facebook.com/people/%D0%9C%D0%91%D0%A4-%D0%A3%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D1%81%D1%8C%D0%BA%D0%B8%D0%B9-%D0%B3%D1%83%D0%BC%D0%B0%D0%BD%D1%96%D1%82%D0%B0%D1%80%D0%BD%D0%B8%D0%B9-%D0%B1%D0%B0%D1%82%D0%B0%D0%BB%D1%8C%D0%BE%D0%BD/100085936770532/"
            target="blank"
          >
            <img
              alt="icon-facebook.svg"
              class="social__img"
              src="/img/icon-facebook.svg"
            />
            <p
              class="social__desc"
            >
              facebook
            </p>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="social"
    >
      <div
        class="social__wrap"
      >
        <h2
          class="social__title"
        >
          соціальні мережі
        </h2>
        <a
          class="social__wrapLink"
          href="https://www.instagram.com/ukrhumbat/?igshid=YmMyMTA2M2Y="
          target="blank"
        >
          <img
            alt="icon-instagram.svg"
            class="social__img"
            src="/img/icon-instagram.svg"
          />
          <p
            class="social__desc"
          >
            instagram
          </p>
        </a>
        <a
          class="social__wrapLink"
          href="https://www.facebook.com/people/%D0%9C%D0%91%D0%A4-%D0%A3%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D1%81%D1%8C%D0%BA%D0%B8%D0%B9-%D0%B3%D1%83%D0%BC%D0%B0%D0%BD%D1%96%D1%82%D0%B0%D1%80%D0%BD%D0%B8%D0%B9-%D0%B1%D0%B0%D1%82%D0%B0%D0%BB%D1%8C%D0%BE%D0%BD/100085936770532/"
          target="blank"
        >
          <img
            alt="icon-facebook.svg"
            class="social__img"
            src="/img/icon-facebook.svg"
          />
          <p
            class="social__desc"
          >
            facebook
          </p>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
